Services/UI/OJS.Services.Ui.Business/Implementations/UsersBusinessService.cs

```diff
@@ -112,17 +140,30 @@ public async Task AddOrUpdateUser(UserProfile user)
             return profile?.Map<UserAuthInfoServiceModel>();
         }
 
-        private bool IsUserAdminOrProfileOwner(string? username)
+        public bool IsUserAdminLecturerOrProfileOwner(string? profileUsername)
```

Comment:
This could be unified with the method IsUserAdminOrProfileOwner. Just pass the allowed roles as args and adjust the implementation. You can make an extension on `currentUser` => `currentUser.IsInRole(roles)`

Reply:
Done.